src/ast/semantic_analyser.cpp

```diff
@@ -756,14 +756,16 @@ void SemanticAnalyser::visit(Unroll &unroll)
   {
     err_ << "unroll maximum value is 20.\n" << std::endl;
   }
-  else if (unroll.var == 0)
+  else if (unroll.var < 1)
   {
     err_ << "unroll minimum value is 1.\n" << std::endl;
   }
 
-  for (Statement *stmt : *unroll.stmts)
-  {
-    stmt->accept(*this);
+  for (int i=0; i < unroll.var; i++) {
```

Comment:
Is the loop necessary here? I can't think of any cases where a set of statements would pass the analyzer once but not multiple times.

Reply:
It's necessary because the semantic_analyser also generates some information which is used later by codegen. For instance, print would not work without this loop, because the code below would only run once

https://github.com/iovisor/bpftrace/blob/master/src/ast/semantic_analyser.cpp#L421-L436

but the code below would expect the above to have run multiple times:

https://github.com/iovisor/bpftrace/blob/master/src/ast/codegen_llvm.cpp#L544
& https://github.com/iovisor/bpftrace/blob/master/src/ast/codegen_llvm.cpp#L579

On a side note, we should probably save printf_id_ in the semantic_analyser as well, to make sure we don't have fmt_str mismatches.

Reply:
I see, that makes sense.